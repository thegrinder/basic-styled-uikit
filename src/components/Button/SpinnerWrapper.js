import styled from 'styled-components';

const SpinnerWrapper = styled.span`
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  margin: auto;
`;

export default SpinnerWrapper;
