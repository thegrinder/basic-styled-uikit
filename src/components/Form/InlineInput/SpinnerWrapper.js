import styled from 'styled-components';
import { rem } from '../../../theme/typography';

const SpinnerWrapper = styled.div`
  position: absolute;
  right: ${({ theme }) => rem(theme, 8)};
  top: 0;
  bottom: 0;
  display: flex;
  align-items: center;
`;

export default SpinnerWrapper;
