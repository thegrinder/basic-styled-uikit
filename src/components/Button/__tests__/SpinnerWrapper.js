import React from 'react';
import { render, cleanup } from '@testing-library/react';


import SpinnerWrapper from '../SpinnerWrapper';

describe('<SpinnerWrapper />', () => {
  afterEach(cleanup);

  it('should render correctly', () => {
    const { container: { firstChild } } = render(
      <SpinnerWrapper />,
    );
    expect(firstChild).toBeDefined();
    expect(firstChild).toMatchSnapshot();
  });

  it('should render <span> tag', () => {
    const { container: { firstChild } } = render(
      <SpinnerWrapper />,
    );
    expect(firstChild.tagName).toEqual('SPAN');
  });
});
