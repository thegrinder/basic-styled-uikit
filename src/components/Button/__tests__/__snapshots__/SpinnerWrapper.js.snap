// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SpinnerWrapper /> should render correctly 1`] = `
<span
  class="SpinnerWrapper-sc-1ln0y9p-0 csgNUt"
/>
`;
